### DISCUSSION TITLE
Getting events from RPH

### DISCUSSION THREAD

**Discussion by isorokuYam** [scripthookvdotnet/scripthookvdotnet#1410] [Q&A]
Hi, is there a way to catch events from RPH (lspdfr exacly)?

**Comment by kagikn** [comment]
A SHVDN script and a RPH plugin have to communicate with each other in a way other than adding a reference to a RPH plugin. Since SHVDN and RPH have their own `AppDomain`s (RPH creates one for each plugin), SHVDN scripts in SHVDN's `AppDomain` can't directly access to data of RPH plugins in RPH's `AppDomain`s, and vice versa.
I think you could achieve the job with some files or some static memory regions with a mutex or a lock flag (SHVDN and RPH create threads for each script/plugin). Maybe some named pipes can do the job?
